### PR TITLE
Add a flag to enable the option to run pre-commit without docker hooks

### PR DESCRIPTION
**Pull request by RotemAmit**
## Related Issues
fixes:https://jira-dc.paloaltonetworks.com/browse/CIAC-8830

## Description
Add the flag `--docker/--no-docker` in order to enable the option to run `demisto-sdk pre-commit` without docker hooks.

### CODE REVIEW

**Comment by JudahSchwartz** [comment]
we dont have a way to enable/disable specific hooks. is that not behavior we want? For example disable pylint-in-docker only but yes to run powershell tests

@eyalpalo @kobymeir

File: demisto_sdk/__main__.py [hunk 1]
```diff
@@ -3463,6 +3463,12 @@ def update_content_graph(
     nargs=-1,
     type=click.Path(exists=True, resolve_path=True, path_type=Path),
 )
+@click.option(
+    "--docker/--no-docker",
+    help="Whether to run docker based hooks or not.",
+    default=False,
```
Comment (eyalpalo) [inline demisto_sdk/__main__.py:3469]: Lets change the default to True as @JudahSchwartz suggested.

File: demisto_sdk/commands/pre_commit/hooks/docker.py [hunk 2]
```diff
@@ -194,15 +194,18 @@ class DockerHook(Hook):
     This class will make common manipulations on commands that need to run in docker
     """
 
-    def prepare_hook(self, files_to_run: Iterable):
+    def prepare_hook(self, files_to_run: Iterable, run_docker_hooks: bool = False):
```
Comment (eyalpalo) [inline demisto_sdk/commands/pre_commit/hooks/docker.py:197]: If we have default value in the command option, do we need this default?

**Review by eyalpalo — CHANGES_REQUESTED** [review]
Nice job!

**Review by eyalpalo — APPROVED** [review]

**Pull request merged**